datatypes/src/primitives/query_rectangle.rs

```diff
@@ -4,37 +4,110 @@ use super::{
 };
 use serde::{Deserialize, Serialize};
 
-/// A spatio-temporal rectangle with a specified resolution
-#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+/// A spatio-temporal rectangle with a specified resolution and the selected bands
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct QueryRectangle<SpatialBounds: AxisAlignedRectangle> {
+pub struct QueryRectangle<SpatialBounds: AxisAlignedRectangle, Selection: QuerySelection> {
     pub spatial_bounds: SpatialBounds,
     pub time_interval: TimeInterval,
     pub spatial_resolution: SpatialResolution,
+    #[serde(default)] // TODO: remove once all clients send this
+    pub selection: Selection,
 }
 
-pub type VectorQueryRectangle = QueryRectangle<BoundingBox2D>;
-pub type RasterQueryRectangle = QueryRectangle<SpatialPartition2D>;
-pub type PlotQueryRectangle = QueryRectangle<BoundingBox2D>;
+pub trait QuerySelection: Clone + Send + Sync + Default /* TOOD: remove */ {}
 
-impl SpatialPartitioned for QueryRectangle<BoundingBox2D> {
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+// TODO: custom deserializer that checks for duplicates(?)
+pub struct BandSelection(Vec<usize>); // TODO: use a large enough Bitmap for a sufficiently large number of bands. Then the `QueryRectangle` can be `Copy` again
+
+impl Default for BandSelection {
+    fn default() -> Self {
```

Comment:
I am not sure if the default shouldn't be "all bands" instead of "the 0th one". But I don't know how to realize this because that would require knowledge about all available bands.